framework/sandstone.cpp

```diff
@@ -3394,6 +3407,23 @@ int main(int argc, char **argv)
                     .range_mode = OutOfRangeMode::Saturate
             }();
             break;
+        case reschedule_option:
+            if (sApp->thread_count < 1) {
+                fprintf(stderr, "%s: --reschedule is only useful with at least 2 threads\n", argv[0]);
+                return EX_USAGE;
+            }
+
+            if (strcmp(optarg, "none") == 0 ) {
+                // Default option, so do nothing
+            } else if (strcmp(optarg, "queue") == 0) {
+                sApp->device_schedule = std::make_unique<QueueDeviceSchedule>(sApp->shmem->cpu_info, num_cpus());
+            } else if (strcmp(optarg, "random") == 0) {
+                sApp->device_schedule = std::make_unique<RandomDeviceSchedule>(sApp->shmem->cpu_info, num_cpus());
```

Comment:
The `-n` restriction hasn't taken effect yet at this point. However, the fact we're processing things immediately means the order of `--cpuset` and `--reschedule` matters. Please make it so they don't.